pkg/jobstore/boltdb/store.go

```diff
@@ -1028,6 +1056,35 @@ func (b *BoltJobStore) appendExecutionHistory(tx *bolt.Tx, updated models.Execut
 	return nil
 }
 
+func (b *BoltJobStore) appendEvaluationHistory(tx *bolt.Tx, eval models.Evaluation) error {
+	historyEntry := models.JobHistory{
+		Type:             models.JobHistoryTypeEvaluationLevel,
+		JobID:            eval.JobID,
+		EvaluationID:     eval.ID,
+		EvaluationStatus: eval.Status,
```

Comment:
A quick skim of the code leads me to believe this value is _always_ set to `pending`. Is there a different value I should use to represent the status of an evaluation?

Reply:
We should update the state to allow differentiating between evaluations that should be processed and those that are fully processed, whether successfully or not. The main reason is to allow the orchestrator to populate the in-memory broker with pending evaluations after a restart:
- `pending` is for new and enqueued evaluations that weren't processed yet
- `complete` is for evaluations that were successfully processed by the scheduler. This doesn't mean that the scheduler was successful in finding nodes, but that it didn't face errors 
- `failed` is for evaluations that schedulers failed to process after `maxReceiveCount` retries
- `cancelled` for deduped evaluations
- `blocked` might for delayed evaluations, but maybe those should be `pending` as well

The important thing is we shouldn't update evaluation state to include in-flight or not, as not worth the state update.

If we are missing those state updates, then this is a good catch and we should create an issue to track this